Delfino/DualCore/SmartWinchDualCore/main.c

```diff
@@ -198,16 +206,12 @@ void main(void)
             GPIO_writePin(DEVICE_GPIO_PIN_LED2, 1); //tic
             //Takes new coordintes XYZ, distance and uplift from previous function
             length4_struct length_out;
-            length_out = coord2ten_sag( (float)modbus_holding_regs[Target_X]/1000.0f,
-                                        (float)modbus_holding_regs[Target_Y]/1000.0f,
-                                        (float)modbus_holding_regs[Target_Z]/1000.0f,
-                                        (float)modbus_holding_regs[Field_Length]/1000.0f,
+            length_out = coord2ten_sag( modbus_holding_regs[Target_X],
```

Comment:
i think you should use an if(dip_switch.BIT7) to disable tension-math just in case it doesnt work.

also, what will happen if the math fail? does it check for valid flag?

what will happen if it is invalid? does it revert back to static-math?

Reply:
The flag is there, but it will activate if an invalid coordinate is given etc. Pythagorean maths can be activated inside the maths function and I don't know how you want to handle an invalid coordinate

Reply:
if invalid coordinate was given, maybe it can:
-ignore the coordinate and don't change the target_length
-revert back to previous valid coordinate